File: ProjeAPI/Models/Mekan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace ProjeAPI.Models
{
    public class Mekan
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int MekanID { get; set; }

        [Required(ErrorMessage = "Mekan adı zorunludur.")]
        [StringLength(100)]
        public string MekanAdi { get; set; } = string.Empty;

        [Required(ErrorMessage = "Şehir zorunludur.")]
        [StringLength(50)]
        public string Sehir { get; set; } = string.Empty;

        [Column(TypeName = "decimal(3,2)")]
        public decimal ZiyaretciPuani { get; set; }

        public DateTime EklenmeTarihi { get; set; } = DateTime.Now;
    }
}

File: proje_mobile/lib/Mekan.dart
class Mekan {
  final int mekanID;
  final String mekanAdi;
  final String sehir;
  final double ziyaretciPuani;

  Mekan({
    required this.mekanID,
    required this.mekanAdi,
    required this.sehir,
    required this.ziyaretciPuani,
  });

  factory Mekan.fromJson(Map<String, dynamic> json) {
    return Mekan(
      mekanID: json['mekanID'] ?? json['MekanID'] ?? 0,
      mekanAdi: json['mekanAdi'] ?? json['MekanAdi'] ?? '',
      sehir: json['sehir'] ?? json['Sehir'] ?? '',
      ziyaretciPuani: (json['ziyaretciPuani'] ?? json['ZiyaretciPuani'] ?? 0.0)
          .toDouble(),
    );
  }
}
